### PR TITLE
[Serving] Extract FlowInvoker to enable easy custom score script

### PR DESCRIPTION
**Pull request by brynn-code**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

# All Promptflow Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](../CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: src/promptflow/promptflow/_sdk/_serving/_errors.py [hunk 1]
```diff
@@ -16,6 +16,24 @@ class MissingRequiredFlowInput(BadRequest):
...
 
 
+class FlowConnectionError(UserErrorException):
+    pass
+
+
+class UnsupportedConnectionProvider(FlowConnectionError):
+    def __init__(self, provider):
+        super().__init__(
+            message_format="Unsupported connection provider {provider}, "
```
Comment (crazygao) [inline src/promptflow/promptflow/_sdk/_serving/_errors.py:26]: if provider is Callable, there may be exceptions here?

Comment (brynn-code) [reply src/promptflow/promptflow/_sdk/_serving/_errors.py:26]: what's the meaning by strange?

File: src/promptflow/promptflow/_sdk/_serving/_errors.py [hunk 2]
```diff
@@ -16,6 +16,24 @@ class MissingRequiredFlowInput(BadRequest):
...
+            "supported are 'local', 'azure' and typing.Callable.",
```
Comment (wangchao1230) [inline src/promptflow/promptflow/_sdk/_serving/_errors.py:27]: let's delete azure in this PR as not implemented

File: src/promptflow/promptflow/_sdk/_serving/flow_invoker.py [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+        self.executor = FlowExecutor.create(
+            flow_file=self.flow_entity.path,
+            working_dir=self.flow_entity.code,
+            connections=self.connections,
+            raise_ex=True,
+        )
+        self.executor.enable_streaming_for_llm_flow(self.streaming)
+        logger.info("Promptflow executor initiated successfully.")
+
+    def invoke(self, data):
```
Comment (wangchao1230) [inline src/promptflow/promptflow/_sdk/_serving/flow_invoker.py:77]: give data some type hint. 

may add example of expected data.

File: src/promptflow/promptflow/_sdk/_serving/_errors.py (hunk 1 above)
Comment (brynn-code) [reply src/promptflow/promptflow/_sdk/_serving/_errors.py]: offline synced

**Review by yigao-alter — APPROVED** [review]

**Review by crazygao — APPROVED** [review]

**Review by wangchao1230 — APPROVED** [review]

**Pull request merged**